mucommander-core/src/main/java/com/mucommander/ui/main/FolderPanel.java

```diff
@@ -130,92 +136,101 @@ it calls actionPerformed() each time an item is highlighted with the arrow (UP/D
         this.mainFrame = mainFrame;
         
         // No decoration for this panel
-        setBorder(null);
-
-        JPanel locationPanel = new JPanel(new GridBagLayout());
-        GridBagConstraints c = new GridBagConstraints();
-        c.fill = GridBagConstraints.HORIZONTAL;
-        c.gridy = 0;
-
-        // Create and add drive button
-        this.driveButton = new DrivePopupButton(this);
-        c.weightx = 0;
-        c.gridx = 0;        
-        locationPanel.add(driveButton, c);
-
-        // Create location text field
-        this.locationTextField = new LocationTextField(this);
-
-        // Give location field all the remaining space until the PoupupsButton
-        c.weightx = 1;
-        c.gridx = 1;
-        // Add some space between drive button and location combo box (none by default)
-        c.insets = new Insets(0, 4, 0, 0);
-        locationPanel.add(locationTextField, c);
-
-        add(locationPanel, BorderLayout.NORTH);
-
-        // Initialize quick lists
-    	fileTablePopups = new QuickList[]{
-    			new ParentFoldersQL(this),
-    			new RecentLocationsQL(this),
-    			new RecentExecutedFilesQL(this),
-    			new BookmarksQL(this),
-    			new RootFoldersQL(this),
-                new TabsQL(this)};
+        panel.setBorder(null);
+
+        JPanel locationPanel = PreloadedJFrame.getJPanel(new GridBagLayout());
+        panel.getTreeLock();
+
+        panel.add(locationPanel, BorderLayout.NORTH);
+
+        new Thread(() -> {
+            GridBagConstraints c = new GridBagConstraints();
+            c.fill = GridBagConstraints.HORIZONTAL;
+            c.gridy = 0;
+
+            // Create and add drive button
+            this.driveButton = new DrivePopupButton(this);
+            c.weightx = 0;
+            c.gridx = 0;
+            locationPanel.add(driveButton, c);
+
+            // Create location text field
+            this.locationTextField = new LocationTextField(this);
+
+            // Give location field all the remaining space until the PoupupsButton
+            c.weightx = 1;
+            c.gridx = 1;
+            // Add some space between drive button and location combo box (none by default)
+            c.insets = new Insets(0, 4, 0, 0);
+            locationPanel.add(locationTextField, c);
+            disableCtrlFocusTraversalKeys(locationTextField);
+            registerCycleThruFolderPanelAction(locationTextField);
+
+            // Allow the location field to change the current directory when a file/folder is dropped on it
+            FileDropTargetListener dropTargetListener = new FileDropTargetListener(this, true);
+            new DropTarget(locationTextField, dropTargetListener);
+            new DropTarget(driveButton, dropTargetListener);
+            locationTextField.addFocusListener(this);
+        }).start();
```

Comment:
did you measure the impact of placing this in a thread? I suppose it's insignificant

Reply:
I don't remember the numbers :/ but showing that drive button took enough time to give my attention into it.